src/Raven.Client/Documents/Session/IGroupByDocumentQuery.cs

```diff
@@ -3,37 +3,57 @@
 
 namespace Raven.Client.Documents.Session
 {
+    /// <summary>
+    /// Perform dynamic group by on collection data. AutoMapReduce index is created if none of the indexes in the database are sufficient to match the query criteria.
```

Comment:
I'm not sure if we should use "perform" word. It suggest the execution of the query which is not the case here. It just creates / defines a group by query.

Instead of "AutoMapReduce" let's use "an auto map-reduce index"

Reply:
"if none of the indexes in the database are sufficient to match the query criteria." -> "if none of _auto_ indexes ..."

We don't scan and use static indexes.

Reply:
I have doubts if we should mention here about creation if auto indexes.